optuna/importance/_mean_decrease_impurity.py

```diff
@@ -0,0 +1,136 @@
+from collections import OrderedDict
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+from typing import ValuesView
+
+import numpy
+
+from optuna._experimental import experimental
+from optuna.distributions import BaseDistribution
+from optuna.distributions import CategoricalDistribution
+from optuna.importance._base import _get_distributions
+from optuna.importance._base import _get_study_data
+from optuna.importance._base import BaseImportanceEvaluator
+from optuna.study import Study
+
+try:
+    from sklearn.compose import ColumnTransformer
+    from sklearn.ensemble import RandomForestRegressor
+    from sklearn.preprocessing import OneHotEncoder
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    _available = False
+
+
+@experimental("1.5.0")
+class MeanDecreaseImpurityImportanceEvaluator(BaseImportanceEvaluator):
+    """Mean Decrease Impurity (MDI) parameter importance evaluator.
+
+    This evaluator fits a random forest that predicts objective values given hyperparameter
+    configurations. Feature importances are then computed using MDI.
```

Comment:
How about adding a reference to MDI for users not familiar with feature importance?
Or, I think a link to the [RandomForest.feature_importances_](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html#sklearn.ensemble.RandomForestClassifier.feature_importances_) will be helpful.

Reply:
Thanks for the suggestion, that makes sense. I added a reference to the existing note in https://github.com/optuna/optuna/pull/1253/commits/8f7a9e1ea0e62eb45a19e4c0d8ac380a224bb0b2. What do you think?

Reply:
SGTM. Thank you!